cl/recap/tasks.py

```diff
@@ -564,13 +566,16 @@ def process_recap_attachment(
     self: Task,
     pk: int,
     tag_names: Optional[List[str]] = None,
-) -> Optional[Tuple[int, str]]:
+    return_rds_affected: bool = False,
+) -> Optional[Tuple[int, str] | list[RECAPDocument]]:
     """Process an uploaded attachment page from the RECAP API endpoint.
 
     :param self: The Celery teask
     :param pk: The primary key of the processing queue item you want to work on
     :param tag_names: A list of tag names to add to all items created or
     modified in this function.
+    :param return_rds_affected: Whether to return a list of attachments
+    documents that were added or not.
```

Comment:
I can see you're trying to be surgical here and make the smallest change possible, but it's better to make a big change than to pile on weird code that changes the return type based on the args passed. 

With that in mind, what'd suggest is simply updating the return so it always returns everything:

```
-> Optional[Tuple[int, str, list[RECAPDocument]]]:
```

That'll require updating anywhere that uses the return types from this function, but that shouldn't be too hard, right?

Reply:
Changed, great thanks. This is good advice! sometimes I'm not sure if it's better to do a major refactor or try to not modify much of the existing method, now I know that it's better to make big changes to avoid fragmented code.

The good thing is that they were not necessary further changes since process_recap_attachment returns are not being used in any other place than the new one.

Reply:
Unless we're doing a hot fix or it's really urgent, always write the best code you can. I try to never compromise on code quality. We're in this for the long haul. 

(Some bugs are just too big to prioritize though, I'll admit that.)